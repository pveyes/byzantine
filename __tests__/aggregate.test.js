const aggregate = require('../aggregate');

test('aggregates single coverages', () => {
  const coverages = [
    {
      statements: {
        covered: 8,
        all: 12,
      },
      branches: {
        covered: 0,
        all: 0,
      },
      functions: {
        covered: 0,
        all: 2,
      },
    }
  ];

  expect(aggregate(coverages)).toMatchSnapshot();
});

test('aggregates multiple coverages', () => {
  const coverages = [
    {
      statements: {
        covered: 14,
        all: 27,
      },
      branches: {
        covered: 5,
        all: 10,
      },
      functions: {
        covered: 0,
        all: 0,
      },
    },
    {
      statements: {
        covered: 8,
        all: 12,
      },
      branches: {
        covered: 0,
        all: 0,
      },
      functions: {
        covered: 0,
        all: 2,
      },
    }
  ];

  expect(aggregate(coverages)).toMatchSnapshot();
});

test('empty coverages', () => {
  const result = aggregate([]);
  expect(result).toMatchSnapshot();
})
