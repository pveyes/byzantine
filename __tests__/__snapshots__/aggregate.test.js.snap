// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aggregates multiple coverages 1`] = `
Object {
  "branches": 50,
  "functions": 0,
  "statements": 56.41025641025641,
}
`;

exports[`aggregates single coverages 1`] = `
Object {
  "branches": 100,
  "functions": 0,
  "statements": 66.66666666666666,
}
`;

exports[`empty coverages 1`] = `
Object {
  "branches": null,
  "functions": null,
  "statements": null,
}
`;
